Karumi/Shot
# Compare screenshots

### :pushpin: References
* **Issue:** Fixes #6 

### :tophat: What is the goal?

Update the project to be able to compare images and get a first version of the plugin comparing real screenshots generated by Facebook's jar.

### How is it being implemented?

This is the list of steps needed to implement this task:

* Add the libraries needed to parse the information generated by the Facebook library. In this case a library to parse xml and other to work with images.

```
compile 'org.scala-lang.modules:scala-xml_2.12:1.0.6'
compile 'com.sksamuel.scrimage:scrimage-core_2.12:2.1.8'
``` 

* Add a new trait to be able to read test resource files named ``Resources``.
* Create the abstractions needed to compare images, parse xml and read files being able to write unit tests easily.
* Implement the XML parser and cover it with integration tests.
* Implement a very first version of the image comparator we will improve in a future PR. It was not so hard to implement because the generated images use tiles. Comparisions are performed in parallel.
* Add some feedback using the ``Console`` abstraction so we can know what's going on.
* Compare images from the main ``Shot`` method and implement the logic needed to show some feedback to the user based on a real comparation.
* Update our model to representing some errors we can handle easily using pattern matching.

### How can it be tested?

Checkout this branch and execute the following commands:

```
./gradlew install
cd shot-consumer
./gradlew executeScreenshotTests
```
You should see some feedback like this:

<img width="996" alt="screen shot 2017-07-21 at 18 09 58" src="https://user-images.githubusercontent.com/4030704/28472181-d96d417c-6e3f-11e7-958a-ae103b5d0cf6.png">

### Next steps

Issue #10 & #5
